gufe/protocols/protocoldag.py

```diff
@@ -345,9 +352,67 @@ def _from_dict(cls, dct: dict):
         return cls(**dct)
 
 
+class ReproduceOldBehaviorStorageManager(StorageManager):
+    # Default behavior has scratch at {dag_label}/scratch/{unit_label} and
+    # shared at {dag_label}/{unit_label}. This little class makes changes
+    # that get us back to the original behavior of this class: scratch at
+    # {dag_label}/scratch_{unit_label} and shared at
+    # {dag_label}/shared_{unit_label}.
+    def _scratch_loc(self, dag_label, unit_label, attempt):
+        return (
+            self.scratch_root
+            / f"{dag_label}/scratch_{unit_label}_attempt_{attempt}"
+        )
+
+    def make_label(self, dag_label, unit_label, attempt):
+        return f"{dag_label}/shared_{unit_label}_attempt_{attempt}"
+
+    @classmethod
+    def from_old_args(
+        cls,
+        shared_basedir: PathLike,
+        scratch_basedir: PathLike, *,
+        keep_shared: bool = False,
+        keep_scratch: bool = False,
+    ):
+        """
+        Create an new storage manager based on the old execute_DAG args.
+
+        Parameters
+        ----------
+        shared_basedir : Path
+            Filesystem path to use for shared space that persists across whole DAG
+            execution. Used by a `ProtocolUnit` to pass file contents to dependent
+            class:``ProtocolUnit`` instances.
+        scratch_basedir : Path
+            Filesystem path to use for `ProtocolUnit` `scratch` space.
+        keep_shared : bool
+            If True, don't remove shared directories for `ProtocolUnit`s after
+            the `ProtocolDAG` is executed.
+        keep_scratch : bool
+            If True, don't remove scratch directories for a `ProtocolUnit` after
+            it is executed.
+        """
+        # doing this here makes it easier to test than putting in
+        # execute_DAG
+        shared_basedir = Path(shared_basedir)
+        shared = FileStorage(shared_basedir.parent)
+        storage_manager = cls(
+            scratch_root=scratch_basedir,
+            shared_root=shared,
+            permanent_root=shared,
+            keep_scratch=keep_scratch,
+            keep_shared=keep_shared,
+            keep_staging=True,
+            keep_empty_dirs=True,
+            staging=Path(""),  # use the actual directories as the staging
+        )
+        return storage_manager
+
+
 def execute_DAG(protocoldag: ProtocolDAG, *,
```

Comment:
are you keeping this for comparison for the PR?  I don't think we need to keep the old behaviour around

Reply:
I think we told Joe that the API was stable enough that he could use it in workflows. So yeah, I spent **a lot** of time ensuring that there would be a code path that doesn't break the current API.